contrib/endpoints/src/api_manager/quota_control.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+//
+#include <iostream>
+
+#include "contrib/endpoints/src/api_manager/cloud_trace/cloud_trace.h"
+#include "contrib/endpoints/src/api_manager/quota_control.h"
+#include "google/protobuf/stubs/status.h"
+
+using ::google::api_manager::utils::Status;
+using ::google::protobuf::util::error::Code;
+
+namespace google {
+namespace api_manager {
+
+void QuotaControl(std::shared_ptr<context::RequestContext> context,
+                  std::function<void(Status status)> continuation) {
+  std::shared_ptr<cloud_trace::CloudTraceSpan> trace_span(
+      CreateSpan(context->cloud_trace(), "QuotaControl"));
+
+  if (!context->service_context()->service_control()) {
+    TRACE(trace_span) << "Service control check is not needed";
```

Comment:
Sorry,  my previous comment is for the message  for trace_space.

we don't need a flag in service config to enable/disable quota.   if there is Quota field with metrics, it is enabled.